docs/proposals/image-provenance.md

```diff
@@ -0,0 +1,358 @@
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Overview
+
+Organizations wish to avoid running "unapproved" images.
+
+The exact nature of "approval" is beyond the scope of Kubernetes, but may include reasons like:
+
+ - only run images that are scanned to confirm they do not contain vulnerabilities
+ - only run images that use a "required" base image
+ - only run images that contain binaries which were built from peer reviewed, checked-in source
+   by a trusted compiler toolchain.
+ - only allow images signed by certain public keys. 
+
+ - etc...
+
+Goals of the design include:
+* Block creation of pods that would cause "unapproved" images to run.
+* Make it easy for users or partners to build "image provenance checkers" which check whether images are "approved".
+  *  We expect there will be multiple implementations.
+* Allow users to request an "override" of the policy in a convenient way (subject to the override being allowed).
+  * "overrides" are needed to allow "emergency changes", but need to not happen accidentally, since they may
+    require tedious after-the-fact justification and affect audit controls.
+
+Non-goals include:
+* Encoding image policy into Kubernetes code.
+* Implementing objects in core kubernetes which describe complete policies for what images are approved.
+  * A third-party implementation of an image policy checker could optionally use ThirdPartyResource to store its policy.
+* Kubernetes core code dealing with concepts of image layers, build processes, source repositories, etc.
+  * We expect there will be multiple PaaSes and/or de-facto programming environments, each with different takes on
+    these concepts.  At any rate, Kuberenetes is not ready to be opinionated on these concepts.
+* Sending more information than strictly needed to a third-party service.
+  * Information sent by Kubernetes to a third-party service constitutes an API of Kubernetes, and we want to
+    avoid making these broader than necessary, as it restricts future evolution of Kubernetes, and makes
+    Kubernetes harder to reason about.  Also, excessive information limits cache-ability of decisions.  Caching
+    reduces latency and allows short outages of the backend to be tolerated.
+
+
+Detailed discussion in [Ensuring only images are from approved sources are run](
+https://github.com/kubernetes/kubernetes/issues/22888).
+
+# Implementation
+
+A new admission controller will be added.  That will be the only change.
+
+## Admission controller
+
+An `ImagePolicyWebhook` admission controller will be written.  The admission controller examines all pod objects which are
+created or updated.  It can either admit the pod, or reject it.  If it is rejected, the request sees a `403 FORBIDDEN`
```

Comment:
This description is a little vague in the plurality sense. I think this should work over a set of webhooks rather than one. For example, this Pull Request on GitHub has multiple webhooks that must validate it before it is merged.